src/order_cmd.cpp

```diff
@@ -269,6 +273,100 @@ void Order::MakeLabel(OrderLabelSubType subtype)
 	this->flags = subtype;
 }
 
+std::string Order::ToJSONString() const
+{
+	std::string out;
+	nlohmann::json json;
+	json["packed-data"] = this->Pack();
+
+	json["destination-id"] = this->GetDestination();
+	Station * station = Station::GetIfValid(this->GetDestination());
+	if(station != nullptr)
+		json["destination-name"] = station->cached_name;
+
+	if(this->extra.get() != nullptr){
+		auto& extraJson = json["extra"];
+
+		extraJson["cargo-type-flags"] = this->extra.get()->cargo_type_flags;
+		extraJson["colour"] = this->extra.get()->colour;
+		extraJson["dispatch-index"] = this->extra.get()->dispatch_index;
+		extraJson["xdata"] = this->extra.get()->xdata;
+		extraJson["xdata2"] = this->extra.get()->xdata2;
+		extraJson["xflags"] = this->extra.get()->xflags;
+	}
+
+	json["refit-cargo"] = this->GetRefitCargo();
+	json["wait-time"] = this->GetWaitTime();
+	json["travel-time"] = this->GetTravelTime();
+	json["max-speed"] = this->GetMaxSpeed();
+
+	out = json.dump();
+	return out;
+}
+
+Order Order::FromJSONString(std::string jsonSTR)
+{
+	nlohmann::json json = nlohmann::json::parse(jsonSTR);
+
+	if (!json.contains("packed-data") && json["packed_data"].is_number_integer()) {
+
+		Order errOrder;
+
+		errOrder.MakeLabel(OLST_TEXT);
+		errOrder.SetColour(COLOUR_RED);
+		errOrder.SetLabelText("JSON_ERR: JSON does not contain mandatory 'packed-data' field for this order");
+
+		return errOrder;
+	}
+
+	Order new_order = Order(json.at("packed-data").get<uint64_t>());
+	
+	if (json.contains("destination-id") && json["destination-id"].is_number_integer()) 
+		json["destination-id"].get_to(new_order.dest);
+		
+	
+
+	if (json.contains("extra") && json["extra"].is_object()) {
+		auto &extraJson = json["extra"];
+
+		new_order.AllocExtraInfo();
+
+		if (extraJson.contains("cargo-type-flags") && extraJson["cargo-type-flags"].is_array()) 
```

Comment:
It'd be better to stick to the OpenTTD code style (braces, etc) where practical.